xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -18,202 +18,141 @@
  *
  */
 
+#if defined(TARGET_ANDROID)
+
 #include "RendererMediaCodecSurface.h"
 
-#if defined(TARGET_ANDROID)
 #include "../RenderCapture.h"
-
+#include "settings/MediaSettings.h"
 #include "platform/android/activity/XBMCApp.h"
 #include "DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.h"
 #include "utils/log.h"
+#include <thread>
 
 CRendererMediaCodecSurface::CRendererMediaCodecSurface()
+  : m_bConfigured(false)
+  , m_iRenderBuffer(0)
 {
 }
 
 CRendererMediaCodecSurface::~CRendererMediaCodecSurface()
 {
-  for (int i(0); i < m_NumYV12Buffers; ++i)
+  for (int i(0); i < m_numRenderBuffers; ++i)
     ReleaseBuffer(i);
 }
 
-bool CRendererMediaCodecSurface::RenderCapture(CRenderCapture* capture)
+bool CRendererMediaCodecSurface::Configure(unsigned int width, unsigned int height, unsigned int d_width, unsigned int d_height, float fps, unsigned flags, ERenderFormat format, void *hwPic, unsigned int orientation)
 {
-  capture->BeginRender();
-  capture->EndRender();
+  CLog::Log(LOGNOTICE, "CRendererMediaCodecSurface::Configure");
+
+  m_sourceWidth = width;
+  m_sourceHeight = height;
+  m_renderOrientation = orientation;
+
+  // Save the flags.
+  m_iFlags = flags;
+  m_format = format;
+
+  // Calculate the input frame aspect ratio.
+  CalculateFrameAspectRatio(d_width, d_height);
+  SetViewMode(CMediaSettings::GetInstance().GetCurrentVideoSettings().m_ViewMode);
+  ManageRenderArea();
+
+  m_bConfigured = true;
+
+  for (int i = 0; i < m_numRenderBuffers; ++i)
+    m_buffers[i].hwPic = 0;
+
   return true;
 }
 
-void CRendererMediaCodecSurface::AddVideoPictureHW(VideoPicture &picture, int index)
+CRenderInfo CRendererMediaCodecSurface::GetRenderInfo()
 {
-  ReleaseBuffer(index);
-  YUVBUFFER &buf = m_buffers[index];
-  buf.hwDec = picture.hwPic ? static_cast<CDVDMediaCodecInfo*>(picture.hwPic)->Retain() : nullptr;
+  CRenderInfo info;
+  info.formats.push_back(RENDER_FMT_BYPASS);
```

Comment:
What do you mean?
I return the question: What that's the advantage of returning a wrong RENDER_FMT_BYPASS vs. the correct RENDER_FMT_MEDIACODECSURFACE ?